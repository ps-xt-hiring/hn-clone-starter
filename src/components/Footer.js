import React from 'react';
import PropTypes from 'prop-types';

function Footer(props) {
  return (
    props.articlesLength ? <div className="click-more" onClick={props.handleMoreClick}>More</div> : ''
  );
}

Footer.propTypes = {
  handleMoreClick: PropTypes.func.isRequired,
  articlesLength: PropTypes.number.isRequired,
};

export default Footer;
