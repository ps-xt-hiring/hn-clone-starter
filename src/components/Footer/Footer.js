// component to show footer

import React from "react";
function Footer(props) {
  return <footer>{props.children}</footer>;
}
export default Footer;
