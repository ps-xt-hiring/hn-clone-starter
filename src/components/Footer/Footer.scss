footer {
  background-color: #f6f6ef;
  padding-top: 5px;
  padding-bottom: 5px;
}
