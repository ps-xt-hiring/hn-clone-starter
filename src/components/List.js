import React from "react";
import PropTypes from 'prop-types';
import {getTime,getUrl} from '../utils/util'
import { Item, ExternalLink, HideLink ,ByText} from "../styles/StoryStyles";
import {StaticText} from '../constant/StaticText'

const List = props => {
  const handleUpVote = () => {
    props.handleUpVote(props.objectID);
  };
  const hideFeed = () => {
    props.hideFeed(props.objectID);
  };
  return (
    <Item isOdd={props.isOdd} key={props.objectID}>
      <span>{props.item.num_comments}</span>
      <span>
        {" "}
        <a href="#" onClick={handleUpVote}>
          <img src="/upvote.png" width={20} alt="upvote"></img>
        </a>
      </span>
      <span>{props.item.points}</span>
      <span>{props.item.title}</span>
      <ExternalLink href={props.item.url}>
        {props.item.url ? getUrl(props.item.url) : null}
      </ExternalLink>
      <ByText>{StaticText.by}:</ByText>
      <span> {props.item.author}</span>
      <span>{getTime(props.item.created_at)}</span>
      <HideLink>
        <a href="#" onClick={hideFeed}>
          [{StaticText.hide}]
        </a>
      </HideLink>
    </Item>
  );
};

List.propTypes={

    item : PropTypes.object,
    handleProductUpVote : PropTypes.func,
    hideFeed : PropTypes.func

}
export default List;
