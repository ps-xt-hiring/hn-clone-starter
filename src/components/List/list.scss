.list-wrapper {
  display: flex;
  align-items: center;
  font-size: 11px;
  padding: 0;
  margin: 0;
  flex-wrap: wrap;

  .item {
    list-style-type: none;
    padding: 0 10px;

    &:nth-child(3) {
      font-weight: 600;
      word-break: break-word;
    }
  }
}
