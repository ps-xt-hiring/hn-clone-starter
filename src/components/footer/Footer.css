.footer-layout {
  width: 15%;
  text-align: right;
}

.footer-btn {
  border: none;
  margin: 10px;
  color: rgb(255, 89, 0);
  background-color: transparent;
  font-size: 0.8em;
  font-weight: 500;
  cursor: pointer;
}

.footer-btn:focus {
  outline: none;
}
