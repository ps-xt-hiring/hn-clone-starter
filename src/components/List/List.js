import React, { Component } from 'react';
import ListItem from '../ListItem/ListItem';
import './List.css';

export default class List extends Component {
  constructor() {
    super();
    this.state = { list: [], currentPage: 0 };
    this.getData = this.getData.bind(this);
    this.hide = this.hide.bind(this);
    this.voteUpdate = this.voteUpdate.bind(this);
  }


  componentDidMount() {
    this.getData();
  }


  getData = () => {
    fetch(
      `http://hn.algolia.com/api/v1/search?tags=front_page&page=${this.state.currentPage}`,
      { method: 'get' },
    )
      .then(resp => resp.json())
      .then((data) => {
        this.setState(prevstate => ({
          list: prevstate.list.concat(data.hits),
          currentPage: prevstate.currentPage + 1,
        }));
      });
  };

  hide = (id) => {
    this.setState({
      list: this.state.list.map((a) => {
        if (a.objectID === id) a.hide = true;
        return a;
      }),
    });
  };

  voteUpdate = (id) => {
    this.setState({
      list: this.state.list.map((a) => {
        if (a.objectID === id) a.points += 1;
        return a;
      }),
    });
  };


  render() {
    return (
      <div className="table-responsive">
        <ListItem
          items={this.state.list}
          hide={this.hide}
          voteUpdate={this.voteUpdate}
        />
        <div className="shw-crsr" role="option" tabIndex="0" aria-selected="true" onClick={this.getData} onKeyPress={this.handleKeyPress}>More</div>
      </div>
    );
  }
}
