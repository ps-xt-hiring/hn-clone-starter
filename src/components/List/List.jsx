import React, {
  Fragment,
  useState,
  useEffect,
  useRef,
} from 'react';
import PropTypes from 'prop-types';
import './list.scss';

let i = 0;

export default function List({
  items = [],
  className = '',
  hideEnable = false,
}) {
  const inputEl = useRef(null);
  const [show, setShow] = useState(true);

  useEffect(() => {
    if (hideEnable) {
      const node = document.createElement('SPAN');
      node.innerHTML = ' [hide]';
      node.setAttribute('role', 'button');
      node.onclick = () => setShow(false);
      inputEl.current.lastChild.appendChild(node);
    }
  }, []);

  return (
    <>
      {show
        && <>
          <ul className={`list-wrapper ${className}`} ref={inputEl}>
            {
              items.map((itm) => {
                i += 1;
                return (
                  <Fragment key={i}>
                    <li className="item">
                      { typeof itm === 'function' ? itm() : itm }
                    </li>
                  </Fragment>
                );
              })
            }
          </ul>
        </>
      }
    </>
  );
}

List.propTypes = {
  items: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.object,
    PropTypes.func,
  ])),
  className: PropTypes.string,
};
