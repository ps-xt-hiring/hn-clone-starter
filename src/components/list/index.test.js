import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import List from './index';
import Footer from '../footer'

const props = {
  feedData: []
};

describe('List component', () => {
  it('shallow renders without crashing', () => {
    shallow(<List {...props} />);
  });

  it('list should not change b/w renders', () => {
    const tree = renderer.create(<List {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('list should change on clicking more', () => {
    const tree = renderer.create(<ListItem {...props} />).toJSON();
    const footer = shallow(<Footer />);
    const moreBtn = wrapper.find('footer span');
    moreBtn.simulate('click');
    expect(tree).not.toMatchSnapshot();
  });
});
