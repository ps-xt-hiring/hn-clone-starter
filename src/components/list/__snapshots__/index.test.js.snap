// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component list should not change b/w renders 1`] = `
<main
  className="list-container"
/>
`;
